force-app/main/default/classes/ALLO_AllocationsRecalculateService_TEST.cls

```diff
@@ -620,6 +620,109 @@ public class ALLO_AllocationsRecalculateService_TEST {
         ALLO_UnitTestHelper_TEST.assertPaymentAllocation(pmtAlloc1, pmts[0].Id, 50.01, percentGAU2, gaus[1].Id, 'Payment Allocation For GAU[1]');
     }
 
+     /**
+     * @description confirms that Allocations are no longer defaulted on Payments when sourceAmount = 0 so that Divide by Zero error no longer occurs.
+     *              with 0 for opportunity and allocation amount, this was done to test W-9458922
+     *              Default Allocations Enabled
+     *              Payment Allocations Enabled
+     *              [ALLO_AllocationsRecalculateService.processPaymentAllocations]
+     *              
+     *              Opportunity Amount $0
+     *                  Opportunity Allocation 1 for GAU 1, $0.00
+     *                  Opportunity Allocation 2 for GAU 2, $0.00
+     *                  Payment amount of 0 with no allocations
+     * 
+     *              No allocations created on payment that is autocreated
+     * 
+     * 
+     */ 
+    @isTest 
+    private static void confirmRecalculateServicePreservesWithZeroAmountOpportunity() {
```

Comment:
Great job on this test, we definitely want to make sure this scenario is covered!